presto-product-tests/src/main/java/com/facebook/presto/tests/ImmutableLdapObjectDefinitions.java

```diff
@@ -66,21 +68,39 @@ private static LdapObjectDefinition buildLdapOrganizationObject(String id, Strin
                 .build();
     }
 
-    private static LdapObjectDefinition buildLdapGroupObject(String groupName, String userName)
+    private static LdapObjectDefinition buildLdapGroupObject(String groupName, String userName, Optional<List<String>> childGroupNames)
     {
-        return buildLdapGroupObject(groupName, americaOrgDistinguishedName, userName, asiaOrgDistinguishedName);
+        if (childGroupNames.isPresent()) {
+            return buildLdapGroupObject(groupName, americaOrgDistinguishedName, userName, asiaOrgDistinguishedName, childGroupNames, Optional.of(americaOrgDistinguishedName));
+        }
+        else {
+            return buildLdapGroupObject(groupName, americaOrgDistinguishedName, userName, asiaOrgDistinguishedName,
+                    Optional.empty(), Optional.empty());
+        }
     }
 
     private static LdapObjectDefinition buildLdapGroupObject(String groupName, String groupOrganizationName,
-                                                             String userName, String userOrganizationName)
+            String userName, String userOrganizationName, Optional<List<String>> childGroupNames, Optional<String> childGroupOrganizationName)
     {
-        return LdapObjectDefinition.builder(groupName)
-                .setDistinguishedName(format("cn=%s,%s", groupName, groupOrganizationName))
-                .setAttributes(ImmutableMap.of(
-                        "cn", groupName,
-                        "member", format("uid=%s,%s", userName, userOrganizationName)))
-                .setObjectClasses(Arrays.asList("groupOfNames"))
-                .build();
+        if (childGroupNames.isPresent() && childGroupOrganizationName.isPresent()) {
+            return LdapObjectDefinition.builder(groupName)
+                    .setDistinguishedName(format("cn=%s,%s", groupName, groupOrganizationName))
+                    .setAttributes(ImmutableMap.of(
+                            "cn", groupName,
+                            "member", format("uid=%s,%s", userName, userOrganizationName)))
+                    .setModificationAttributes(getAttributes(childGroupNames.get(), childGroupOrganizationName.get(), member_relation))
```

Comment:
For subgroup relationships, we are only adding a "member" relationship but not "memberOf" relationship. Can you make sure that this one way specification is sufficient?